src/Common/Core/Impl/OS/Win32EnvironmentBlock.cs

```diff
@@ -40,63 +61,24 @@ public class Win32EnvironmentBlock : Dictionary<string, string> {
             return eb;
         }
 
-        public byte[] ToByteArray() {
-            using (MemoryStream ms = new MemoryStream()) {
-                byte[] nulls = { 0, 0 };
-                foreach (var p in this) {
-                    string envData = $"{p.Key}={p.Value}";
-                    byte[] data = Encoding.Unicode.GetBytes(envData);
-                    ms.Write(data, 0, data.Length);
+        private byte[] ToByteArray() {
+            lock (_environment) {
```

Comment:
This doesn't do anything useful now, since nothing else uses `lock` on that object. I'm not sure if it's even safe to do for `ConcurrentDictionary`.

If the only thing you care about is for enumeration to not throw, then you don't need locking here at all - `ConcurrentDictionary` guarantees that enumeration will succeed even if elements are added or removed concurrently. However, it doesn't guarantee anything about which elements might or might not show up. It sounds like in this case we don't actually care? So you can just ditch this.

If you actually need snapshot semantics here, the best way to achieve it is to call `ToArray` on the dictionary. It has its own implementation that will make a snapshot under internal lock.